### PR TITLE
 Add gpg artifact publishing

### PR DESCRIPTION
**Pull request by hallipr**
- After each maven publish step, publish the gpg signed build artifacts
- Move the existing package check in Publish-MavenPacakges.ps1 to after the gpg signing step, so the build always produces a signed artifact

### CODE REVIEW

File: eng/pipelines/partner-release.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ resources:
 variables:
   BuildToolScripts: $(Pipeline.Workspace)/azure-sdk-build-tools/scripts
   Artifacts: $(Pipeline.Workspace)/artifacts
+  GpgArtifacts: $(Pipeline.Workspace)/artifacts-gpg
```
Comment (weshaggard) [inline eng/pipelines/partner-release.yml:28]: This doesn't look like we use it.

**Review by weshaggard — APPROVED** [review]

File: eng/scripts/Publish-MavenPackages.ps1 [hunk 2]
```diff
@@ -62,6 +64,8 @@ Write-Host "Starting GPG signing and publishing"
 foreach ($packageDetail in $packageDetails) {
   Write-Host "GPG signing and publishing package: $($packageDetail.FullyQualifiedName)"
 
+  $deploymentType = ""
```
Comment (weshaggard) [inline eng/scripts/Publish-MavenPackages.ps1:67]: Looks like it this is leftover.

File: eng/pipelines/partner-release.yml (hunk 1 above)
Comment (hallipr) [reply eng/pipelines/partner-release.yml]: Removed.  Thanks

File: eng/scripts/Publish-MavenPackages.ps1 (hunk 2 above)
Comment (hallipr) [reply eng/scripts/Publish-MavenPackages.ps1]: Also removed. Also thanks.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**